docs/development/pigs/pig-008.rst

```diff
@@ -0,0 +1,449 @@
+cases it's useful to simulate counts from a predicted number of counts model. Two
+approaches are possible: simulating a counts map by calling `np.random.poisson()` on
+a predicted number of counts map or sampling of event lists. The latter is addressed
+in a separate PIG (#009). The direct sampling of binned data could be supported
+as following:
+
+.. code::
+
+    # allow counts=None on init
+    dataset = MapDataset(counts=None)
+
+    # to sample a counts map from npred
+    dataset.counts = dataset.simulate_counts(random_seed=0)
+
+    fit = Fit(dataset)
+    fit.optmimize()
+
+
+List of Pull Requests
+=====================
+
+This is a proposal for a list of pull requests implementing the proposed changes,
+ordered by priority:
+
+1. Refactor the current `FluxPointFit` into a `FluxPointsDataset` class
+and make it work with the current `Fit` class. Ensure that the `Fit` class
+supports the old inheritance scheme as well as the new dataset on init.
+Update tests and tutorials (**v0.11**).
+
+2. Refactor the current `MapFit` into the `MapDataset` class. Only support a
+fixed energy dispersion and psf first. Use the current `MapEvaluator` for model
+evaluation, but split out the background evaluation. Update test and tutorials (**v0.11**)
+
+3. Implement the `Datasets` class in `gammapy.utils.fitting.datasets`, add tests
+using multiple `MapDataset`. Change the `Fit` interface to take a `Datasets`
+object or list of `MapDataset` on input (**v0.11**).
+
+4. Enable joint-likelihood analyses by filtering the `Datasets.parameters` for
+unique parameters only. Add a tutorial for joint-likelihood fitting of multiple
+observations (**v0.11**).
+
+5. Implement `MapDataset.setup()` method, using a list of `MapEvaluator`
+objects. Add an `.evaluation_radius()` attribute to all the spatial models.
+Only support a fixed PSF and edisp per model component (**v0.11**).
+
+6. Add support for psf maps to `MapDataset.setup()`. Extend the `.setup()` method
+to look up the correct PSF for the given model component. A `PSFMap` has to be
+passed on `MapDataset.__init__()` (**v0.11**).
+
+7. Add support for energy dispersion maps to `MapDataset`. Extend the `.setup()`
+method to look up the correct EDisp for a given model component. An `EdispMap`
+has to be passed on `MapDataset.__init__()` (**v0.11**).
+
+8. Add tutorial for joint-likelihood fitting of IACT and Fermi-LAT data, based
+on the joint-crab dataset (**v0.12**).
+
+9. Add a `name` attribute to datasets and a `Datasets.__getitem__` method (**v0.12**).
+
+10. Implement dataset serialization to yaml (**v0.12**).
+
+11. Implement datasets serialization to a single fits file (**v0.12**).
+
+12. Add the direct likelihood evaluation to the `FluxPointsDataset`, by
+interpolating flux point likelihood profiles (**v0.12**).
+
+
+
+Alternatives
+============
+Joint-likelihood analyses for multiple instruments are already possible using
```

Comment:
I don't think 3ML is a full alternative.

We need Gammapy by itself to do good IACT data analysis. This requires datasets and joint fit across datasets, where the dataset can be spectra or maps from IACT observations.

Also per-obs background model with individual parameters is very common and needs to be supported. Suggest to mention this specifically as a use case also in the abstract.

I'm not sure there's really a full alternative to this PIG. You could change this section to "Future work" or "Open questions" or something like that, and as part of it share some thoughts that "Gammapy dataset"  is similar to "3ML plugin" and that our focus is on the kinds of datasets used for IACTs (3D maps, 2D maps, 1D spectra, spectral points and likelihood profiles), and that this is pretty much already a full MWL solution, like Sherpa, 3ML, but our motivation was just to do IACT analysis well.

Reply:
I agree there is no real full alternative, so I just decided to remove the `Alternatives` section.